client/src/components/HomePage.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from "react";
+import Header from "./Header";
+import VCard from "./VCard";
+import AddVideo from "./AddVideo";
+import Sort from "./Sort";
+import Footer from "./Footer";
+
+const HomePage = () => {
+  const [videoData, setVideoData] = useState([]);
+  const [sortOrder, setSortOrder] = useState("desc");
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetch("https://video-3.onrender.com/")
+      .then((response) => response.json())
+      .then((data) => {
+        setVideoData(data);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.log(error);
+        setIsLoading(false);
+      });
+  }, []);
+
+  const updateVideoData = (newVideoData) => {
+    fetch(`https://video-3.onrender.com/?order=${sortOrder}`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newVideoData),
+    })
+      .then((response) => response.json())
+      .then((data) => setVideoData(data))
```

Comment:
Small point, but the server could send back a 400. If `response.ok` is false, you know there's an error so you shouldn't call `setVideoData`